pulsar-client-1x-base/pulsar-client-1x/src/main/java/org/apache/pulsar/client/api/MessageListener.java

```diff
@@ -33,6 +33,8 @@
      *
      * This method will only be called once for each message, unless either application or broker crashes.
      *
+     * Implementation should acknowledge messages by calling consumer.acknowledge(msg).
```

Comment:
I am not sure the changes here.

1) I don't think an implementation need to ack messages. *should* here is a bit misleading.

2) there are many methods to ack a message, ack vs ackCumulative, sync vs async. Also it is recommended to call async methods in the message listener otherwise a sync method can block the listener thread.

Reply:
Hm, 1. who will ack messages then? 2. Fair

Reply:
I mean it doesn't have to ack within in the listener implementation. The application can ack the message in other thread, or even the application can batch the messages and ack them together.

Reply:
So, something like "Application is responsible for acking message by calling any of consumer acknowledgement methods" will be better?

Reply:
@Guchman yeah that's better.